x-pack/plugins/security_solution/public/cloud_security_posture/routes.tsx

```diff
@@ -39,11 +40,13 @@ const CloudSecurityPosture = memo(() => {
 
   return (
     // TODO: Finer granularity of this needs to be implemented in the cloud security posture plugin
-    <TrackApplicationView viewId="cloud_security_posture">
-      <SecuritySolutionPageWrapper noPadding noTimeline>
-        <CloudSecurityPostureRouter securitySolutionContext={securitySolutionContext} />
-      </SecuritySolutionPageWrapper>
-    </TrackApplicationView>
+    <PluginWrapper>
```

Comment:
is the name `PluginWrapper` meant to imply wrapping should be done on all of a plugin's routes and not individual pages?

asking because for now that's our intended usage - change a page template layout based on some internal page state, which we cannot lift here. (example states: loading, no data, missing integration)

Reply:
It can be done both wrapping all the routes or a single one; for backwards compatiblity. If you opt-in to manage the layout yourself, you should go with per-route approach.

As for the naming, I am open for suggestions:). Maybe PluginRouteWrapper will work better, mentioning the above in the jsdoc.